### PR TITLE
add cache test suite

### PR DESCRIPTION
**Pull request by xuyangit**

### CODE REVIEW

**Comment by xuyangit** [comment]
Currently, just provide 2 cases.

File: src/main/scala/org/apache/spark/sql/suites/CacheMissSuite.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+object CacheMissSuite extends OapTestSuite with CacheMissConfigSet {
+
+  val table = "store_sales"
+
+  val attr = "ss_customer_sk"
+
+  private def databaseName = {
+    val conf = activeConf
+    conf.getBenchmarkConf(BenchmarkConfig.FILE_FORMAT) match {
+      case "parquet" => "parquet_tpcds_200"
```
Comment (yueawang) [inline src/main/scala/org/apache/spark/sql/suites/CacheMissSuite.scala]: I updated some code, please use 
OapBenchmarkDataBuilder.getDatabase(activeConf.getBenchmarkConf(BenchmarkConfig.FILE_FORMAT))

File: src/main/scala/org/apache/spark/sql/BenchmarkConfig.scala [hunk 2]
```diff
@@ -216,6 +216,21 @@ trait OapStrategyConfigSet extends BenchmarkConfigSelector{
...
+    new BenchmarkConfig()
+      .setBenchmarkConfName("offheap memory only")
+      .setBenchmarkConf(BenchmarkConfig.FILE_FORMAT, "oap")
+      .setSparkConf("spark.memory.offHeap.enabled", "true")
+      .setSparkConf("spark.memory.offHeap.size", "10g")
+      .setSqlConf("spark.sql.oap.oindex.eis.enabled", "false"),
+    new BenchmarkConfig()
+      .setBenchmarkConfName("no offheap used")
+      .setBenchmarkConf(BenchmarkConfig.FILE_FORMAT, "oap")
+      .setSparkConf("spark.memory.offHeap.enabled", "false")
```
Comment (yueawang) [inline src/main/scala/org/apache/spark/sql/BenchmarkConfig.scala:230]: spark.memory.offHeap.enabled => false?
Did not you meet an assertion about it??

**Review by yueawang — COMMENTED** [review]
Thanks Yang, LGTM with few comments.

File: src/main/scala/org/apache/spark/sql/BenchmarkConfig.scala (hunk 2 above)
Comment (xuyangit) [reply src/main/scala/org/apache/spark/sql/BenchmarkConfig.scala:230]: @yueawang ,don't get ur point. I have run this suite last day and no error got. Why assertion?

File: src/main/scala/org/apache/spark/sql/suites/CacheMissSuite.scala (hunk 1 above)
Comment (xuyangit) [reply src/main/scala/org/apache/spark/sql/suites/CacheMissSuite.scala]: Ok,update it later

**Pull request merged**